### ISSUE TITLE
Invoke-IcingaCheckStoragePool 1.4.1 - Error with OperationalStatus

### ISSUE THREAD

**Issue by puseidr** [Icinga/icinga-powershell-plugins#149]
Hi, 

the Invoke-IcingaCheckStoragePool cmdlet is trying to convert a String ("OK") into Int32.

```
PS C:\Users\Administrator> Invoke-IcingaCheckStoragePool
Der Wert "OK" kann nicht in den Typ "System.Int32" konvertiert werden. Fehler: "Die Eingabezeichenfolge hat das
falsche Format."
In C:\Program
Files\WindowsPowerShell\Modules\icinga-powershell-plugins\provider\storage\Get-IcingaStoragePoolInfo.psm1:77 Zeichen:17
+ ...             Add-IcingaHashtableItem -Hashtable $OperationalStatus -Ke ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvalidCastFromStringToInteger

[CRITICAL] Check package "Storage Pools Package" - [CRITICAL] Storage Pool: Operational Status
\_ [CRITICAL] Check package "Storage Pool"
   \_ [CRITICAL] Storage Pool: Operational Status
| 'storage_pool_totalused'=8267.75GB;; 'storage_pool_capacity'=10907GB;; 'storage_pool_freespace'=2639.25GB;;
2
```

the output of "Get-StoragePool looks like this:
```
PS C:\Users\Administrator> Get-StoragePool

FriendlyName OperationalStatus HealthStatus IsPrimordial IsReadOnly     Size AllocatedSize
------------ ----------------- ------------ ------------ ----------     ---- -------------
Storage Pool OK                Healthy      False        False      10.65 TB       8.07 TB
Primordial   OK                Healthy      True         False      22.12 TB      10.66 TB
```

This happens on a Windows Server 2019 Standard Edition with icinga powershell plugins 1.4.1

**Comment by LordHepipud** [comment]
This is an interesting issue I can't seem to reproduce. We are not using `Get-StoragePool`, but `Get-IcingaWindowsInformation -ClassName MSFT_StoragePool -Namespace 'Root/Microsoft/Windows/Storage'`.

The operational status in this case is an integer, reflecting the state. It is just printed string later on. Does this happen on other machines as well or just on this one?

**Comment by puseidr** [comment]
we have this problem on all servers working with storage spaces or S2D

**Comment by LordHepipud** [comment]
Could you please run this command:

```powershell
Get-IcingaWindowsInformation -ClassName MSFT_StoragePool -Namespace 'Root/Microsoft/Windows/Storage'
```

And give me a short summary of the result? Especially the `Operational Status` would be interesting in this matter. Thanks!

**Comment by puseidr** [comment]
here is the output of an S2D Cluster-Node:
```
PS C:\> Get-IcingaWindowsInformation -ClassName MSFT_StoragePool -Namespace 'Root/Microsoft/Windows/Storage'

FriendlyName     OperationalStatus HealthStatus IsPrimordial IsReadOnly     Size AllocatedSize
------------     ----------------- ------------ ------------ ----------     ---- -------------
Primordial       OK                Healthy      True         False      35.15 TB      34.93 TB
Primordial       OK                Healthy      True         False      35.15 TB      34.93 TB
S2D on HVCluster OK                Healthy      False        False      34.92 TB      14.03 TB
```

```
ObjectId                          : {1}\\HVCluster\root/Microsoft/Windows/Storage/Providers_v2\SPACES_StoragePool.ObjectId="{3df4ae72-9232-11eb-8f49-806e6f6e6963}:SP:{3df4ae71-9232-11eb-8f49-806e6f6e6963}"
PassThroughClass                  :
PassThroughIds                    :
PassThroughNamespace              :
PassThroughServer                 :
UniqueId                          : {3df4ae71-9232-11eb-8f49-806e6f6e6963}
AllocatedSize                     : 38402187591680
ClearOnDeallocate                 : False
EnclosureAwareDefault             : False
FaultDomainAwarenessDefault       : 1
FriendlyName                      : Primordial
HealthStatus                      : 0
IsClustered                       : False
IsPowerProtected                  : False
IsPrimordial                      : True
IsReadOnly                        : False
LogicalSectorSize                 :
MediaTypeDefault                  : 0
Name                              :
OperationalStatus                 : {2}
OtherOperationalStatusDescription :
OtherUsageDescription             :
PhysicalSectorSize                :
ProvisioningTypeDefault           : 2
ReadOnlyReason                    : 1
RepairPolicy                      : 3
ResiliencySettingNameDefault      : Mirror
RetireMissingPhysicalDisks        : 1
Size                              : 38647004250112
SupportedProvisioningTypes        : {1, 2}
SupportsDeduplication             : True
ThinProvisioningAlertThresholds   : {70}
Usage                             : 1
Version                           : 22
WriteCacheSizeDefault             : 18446744073709551615
WriteCacheSizeMax                 : 18446744073709551614
WriteCacheSizeMin                 : 0
PSComputerName                    :

ObjectId                          : {1}\\HVCluster\root/Microsoft/Windows/Storage/Providers_v2\SPACES_StoragePool.ObjectId="{5ec1c4b3-5439-4733-8c3b-b5e59ad946a4}:SP:{9bd339ca-5086-4012-b70d-1b0a94381e46}"
PassThroughClass                  :
PassThroughIds                    :
PassThroughNamespace              :
PassThroughServer                 :
UniqueId                          : {9bd339ca-5086-4012-b70d-1b0a94381e46}
AllocatedSize                     : 38402187591680
ClearOnDeallocate                 : False
EnclosureAwareDefault             : False
FaultDomainAwarenessDefault       : 1
FriendlyName                      : Primordial
HealthStatus                      : 0
IsClustered                       : True
IsPowerProtected                  : False
IsPrimordial                      : True
IsReadOnly                        : False
LogicalSectorSize                 :
MediaTypeDefault                  : 0
Name                              :
OperationalStatus                 : {2}
OtherOperationalStatusDescription :
OtherUsageDescription             :
PhysicalSectorSize                :
ProvisioningTypeDefault           : 2
ReadOnlyReason                    : 1
RepairPolicy                      : 3
ResiliencySettingNameDefault      : Mirror
RetireMissingPhysicalDisks        : 1
Size                              : 38647004250112
SupportedProvisioningTypes        : {2}
SupportsDeduplication             : True
ThinProvisioningAlertThresholds   : {70}
Usage                             : 1
Version                           : 22
WriteCacheSizeDefault             : 18446744073709551615
WriteCacheSizeMax                 : 18446744073709551614
WriteCacheSizeMin                 : 0
PSComputerName                    :

ObjectId                          : {1}\\HVCluster\root/Microsoft/Windows/Storage/Providers_v2\SPACES_StoragePool.ObjectId="{5ec1c4b3-5439-4733-8c3b-b5e59ad946a4}:SP:{afff8b57-b7f2-4309-8d03-e8d9bcb8223b}"
PassThroughClass                  :
PassThroughIds                    :
PassThroughNamespace              :
PassThroughServer                 :
UniqueId                          : {afff8b57-b7f2-4309-8d03-e8d9bcb8223b}
AllocatedSize                     : 15425911914496
ClearOnDeallocate                 : False
EnclosureAwareDefault             : False
FaultDomainAwarenessDefault       : 3
FriendlyName                      : S2D on HVCluster
HealthStatus                      : 0
IsClustered                       : True
IsPowerProtected                  : True
IsPrimordial                      : False
IsReadOnly                        : False
LogicalSectorSize                 : 4096
MediaTypeDefault                  : 0
Name                              :
OperationalStatus                 : {2}
OtherOperationalStatusDescription :
OtherUsageDescription             : Reserved for S2D
PhysicalSectorSize                : 4096
ProvisioningTypeDefault           : 2
ReadOnlyReason                    : 1
RepairPolicy                      : 3
ResiliencySettingNameDefault      : Mirror
RetireMissingPhysicalDisks        : 3
Size                              : 38396818882560
SupportedProvisioningTypes        : {2}
SupportsDeduplication             : True
ThinProvisioningAlertThresholds   : {70}
Usage                             : 1
Version                           : 22
WriteCacheSizeDefault             : 18446744073709551615
WriteCacheSizeMax                 : 18446744073709551614
WriteCacheSizeMin                 : 0
PSComputerName                    :
```




and with a normal Storage Space Server:

```
ObjectId                          : {1}\\SRV-INFRA01\root/Microsoft/Windows/Storage/Providers_v2\SPACES_StoragePool.Obj
                                    ectId="{8e63c19e-9644-11e8-9196-806e6f6e6963}:SP:{46bdef1c-2013-43fa-a4a6-52c07379a
                                    906}"
PassThroughClass                  :
PassThroughIds                    :
PassThroughNamespace              :
PassThroughServer                 :
UniqueId                          : {46bdef1c-2013-43fa-a4a6-52c07379a906}
AllocatedSize                     : 8877428965376
ClearOnDeallocate                 : False
EnclosureAwareDefault             : False
FaultDomainAwarenessDefault       : 1
FriendlyName                      : Storage Pool
HealthStatus                      : 0
IsClustered                       : False
IsPowerProtected                  : False
IsPrimordial                      : False
IsReadOnly                        : False
LogicalSectorSize                 : 512
MediaTypeDefault                  : 0
Name                              :
OperationalStatus                 : {2}
OtherOperationalStatusDescription :
OtherUsageDescription             :
PhysicalSectorSize                : 4096
ProvisioningTypeDefault           : 2
ReadOnlyReason                    : 1
RepairPolicy                      : 3
ResiliencySettingNameDefault      : Mirror
RetireMissingPhysicalDisks        : 1
Size                              : 11711302074368
SupportedProvisioningTypes        : {1, 2}
SupportsDeduplication             : True
ThinProvisioningAlertThresholds   : {70}
Usage                             : 1
Version                           : 19
WriteCacheSizeDefault             : 18446744073709551615
WriteCacheSizeMax                 : 18446744073709551614
WriteCacheSizeMin                 : 0
PSComputerName                    :

ObjectId                          : {1}\\SRV-INFRA01\root/Microsoft/Windows/Storage/Providers_v2\SPACES_StoragePool.Obj
                                    ectId="{8e63c19e-9644-11e8-9196-806e6f6e6963}:SP:{8e63c19d-9644-11e8-9196-806e6f6e6
                                    963}"
PassThroughClass                  :
PassThroughIds                    :
PassThroughNamespace              :
PassThroughServer                 :
UniqueId                          : {8e63c19d-9644-11e8-9196-806e6f6e6963}
AllocatedSize                     : 11720648032256
ClearOnDeallocate                 : False
EnclosureAwareDefault             : False
FaultDomainAwarenessDefault       : 1
FriendlyName                      : Primordial
HealthStatus                      : 0
IsClustered                       : False
IsPowerProtected                  : False
IsPrimordial                      : True
IsReadOnly                        : False
LogicalSectorSize                 :
MediaTypeDefault                  : 0
Name                              :
OperationalStatus                 : {2}
OtherOperationalStatusDescription :
OtherUsageDescription             :
PhysicalSectorSize                :
ProvisioningTypeDefault           : 2
ReadOnlyReason                    : 1
RepairPolicy                      : 3
ResiliencySettingNameDefault      : Mirror
RetireMissingPhysicalDisks        : 1
Size                              : 24321519247360
SupportedProvisioningTypes        : {1, 2}
SupportsDeduplication             : True
ThinProvisioningAlertThresholds   : {70}
Usage                             : 1
Version                           : 22
WriteCacheSizeDefault             : 18446744073709551615
WriteCacheSizeMax                 : 18446744073709551614
WriteCacheSizeMin                 : 0
PSComputerName                    :
```

**Comment by LordHepipud** [comment]
Thanks for providing these information. I'm not sure on why there is a problem.

Could you please try this one in addition?

```powershell
(Get-IcingaWindowsInformation -ClassName MSFT_StoragePool -Namespace 'Root/Microsoft/Windows/Storage').OperationalStatus
```

This should print out the actual content of the variable, because something seems to be missing here.

**Comment by puseidr** [comment]
here is the output of an S2D Cluster-Node (Server 2019 Datacenter (german)):

```
PS C:\Windows\system32> use-icinga
PS C:\Windows\system32> (Get-IcingaWindowsInformation -ClassName MSFT_StoragePool -Namespace 'Root/Microsoft/Windows/Storage').OperationalStatus
2
2
2
PS C:\Windows\system32>
```

and with a normal Storage Space Server (Server 2019 Standard (german)):

```
PS C:\Windows\system32> use-icinga
PS C:\Windows\system32> (Get-IcingaWindowsInformation -ClassName MSFT_StoragePool -Namespace 'Root/Microsoft/Windows/Storage').OperationalStatus
2
2
```

**Comment by LordHepipud** [comment]
Thank you for the input. I honestly don't understand on where the problem comes from.
The value is `2`. which means `OK` and should be used within the plugin. I don't get on where the `OK` String for comparison is coming from right now.

**Comment by LordHepipud** [comment]
This should be fixed with v1.7.0
